stl/inc/stacktrace

```diff
@@ -316,6 +312,15 @@ private:
         _CATCH_END
     }
 
+    void _Prepare_space(size_type _Size) {
+        // <NOTICE> resize _Vec without value initialization, discarding previously allocated buffer if there is.
+        // <NOTICE> currently I cannot think of a better name, and the comment rely on the final decision of _Current.
+
+        // static_assert(is_implicit_lifetime_v<stacktrace_entry>); TRANSITION, not implemented
+        const auto _Newvec = _Frames._Getal().allocate(_Size);
+        _Frames._Change_array(_Newvec, _Size, _Size);
+    }

```

Comment:
I isolated uninitialized-resize logic to make it more detachable; however removing this optimization will make attempts to improve efficiency a trade-off problem(whether to accept worse efficiency in large cases)

Reply:
As `allocator_type` is not necessarily `std::allocator`, is "start_lifetime_as_array" machinery needed here?

Reply:
> It seems this will still involve UB when `allocator_type` is not `std::allocator`... Am I missing something? Does "Allocator"::`allocate` implicitly create objects? [(implicit creation)](https://en.cppreference.com/w/cpp/language/object#Object_creation:~:text=Call%20to%20std%3A%3Aallocator%3A%3Aallocate) [(basic_stacktrace::allocator_type)](https://en.cppreference.com/w/cpp/utility/basic_stacktrace#:~:text=The%20type%20must%20meet%20the%20requirements%20of%20Allocator)

Oh, I had missed that fact that `std::allocator<T>::allocate` doesn't implicitly create element objects.

We can still use `::operator new(sizeof(stacktrace_entry) * _Size, static_cast<void*>(_Unfancy(_Newvec)));` to implicitly create element objects. The newly created `stacktrace_entry` objects will have indeterminate values, but I think we can assume that subsequent operations will rewrite them.

Reply:
Done! Hope there is no other UB(s) in this function